src/DssSpell.t.sol

```diff
@@ -7,285 +7,349 @@ import "dss-interfaces/Interfaces.sol";
+        assertEq(vest.res(29), 1);
+        assertEq(vest.tot(29), 412_085 * WAD);
+        assertEq(vest.rxd(29), 0);
 
-    function testPayouts() public {
-        address sne = wallets.addr("SNE_WALLET");
-        address tech = wallets.addr("TECH_WALLET");
-        address oraGas = wallets.addr("ORA_GAS");
-        address oraGasEmerg = wallets.addr("ORA_GAS_EMERGENCY");
 
-        uint256 snebal = dai.balanceOf(sne);
-        uint256 techbal = dai.balanceOf(tech);
-        uint256 oraGasbal = dai.balanceOf(oraGas);
-        uint256 oraGasEmergbal = dai.balanceOf(oraGasEmerg);
+        // // Give admin powers to Test contract address and make the vesting unrestricted for testing
+        hevm.store(
+            address(vest),
+            keccak256(abi.encode(address(this), uint256(1))),
+            bytes32(uint256(1))
+        );
+        vest.unrestrict(28);
+        vest.unrestrict(29);
+        // //
 
-        vote(address(spell));
-        scheduleWaitAndCast(address(spell));
-        assertTrue(spell.done());
+        hevm.warp(JUL_31_2022);
+        uint256 prevBalanceSNE  = dai.balanceOf(wallets.addr("SNE_WALLET"));
+        uint256 prevBalanceSF  = dai.balanceOf(wallets.addr("SF_WALLET"));
+
+        uint256 vestedSNE = vest.accrued(28);
+        assertEq(vestedSNE, 214583 * WAD);
+        uint256 vestedSF = vest.accrued(29);
+        assertEq(vestedSF, 412085 * WAD);
 
-        assertEq(dai.balanceOf(sne) - snebal, 229_792 * WAD);
-        assertEq(dai.balanceOf(tech) - techbal, 1_069_250 * WAD);
-        assertEq(dai.balanceOf(oraGas) - oraGasbal, 6_966_070 * WAD);
-        assertEq(dai.balanceOf(oraGasEmerg) - oraGasEmergbal, 1_805_407 * WAD);
+        vest.vest(28);
+        vest.vest(29);
+
+        assertEq(dai.balanceOf(wallets.addr("SNE_WALLET")), prevBalanceSNE + vestedSNE);
+        assertEq(dai.balanceOf(wallets.addr("SF_WALLET")), prevBalanceSF + vestedSF);
 
     }
 
-    function testVestDAI() public {
+    // Basically a mirror of the above test but we simulate vesting just before the cast to ensure it is not blocked
+    function testVestDAINotBlocking() public {
         VestAbstract vest = VestAbstract(addr.addr("MCD_VEST_DAI"));
 
         uint streams = vest.ids();
 
+        // // Give admin powers to Test contract address and make the vesting unrestricted for testing
+        hevm.store(
+            address(vest),
+            keccak256(abi.encode(address(this), uint256(1))),
+            bytes32(uint256(1))
+        );
+        vest.unrestrict(24);
+        vest.unrestrict(26);
+        // //
+
+        // Ensure that vested contracts do not block casting of the spell
+        vest.vest(24);
+        vest.vest(26);
+
         vote(address(spell));
         scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
         assertEq(vest.cap(), 1 * MILLION * WAD / 30 days);
-        assertEq(vest.ids(), streams + 6);
+        assertEq(vest.ids(), streams + 2);
 
```

Comment:
I would add the following checks to ensure ids 24 and 26 are yanked at cast time.
```suggestion
        assertEq(vest.fin(24), block.timestamp);
        assertEq(vest.fin(26), block.timestamp);
        assertEq(vest.tot(24), vest.accrued(24));
        assertEq(vest.tot(26), vest.accrued(26));
```